### PR TITLE
mgmt containerinstance, support liveness readiness probe

### PR DESCRIPTION
**Pull request by XiaofeiCao**
# Description

link https://github.com/Azure/azure-sdk-for-java/issues/32265

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-containerinstance/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java [hunk 1]
```diff
@@ -922,6 +922,78 @@ WithContainerInstanceAttach<ParentT> withReadOnlyVolumeMountSetting(
...
+             */
+            interface WithLivenessProbe<ParentT> {
+                /**
+                 * Specifies the container's liveness probe to execute a given command at a given interval.
+                 *
+                 * @param command the command for the probe to execute
+                 * @param probePeriodSeconds the interval at which the command executes
+                 * @return the next stage of the definition
+                 */
+                WithContainerInstanceAttach<ParentT> withLivenessProbeExec(List<String> command, int probePeriodSeconds);
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-containerinstance/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java:940]: I am not sure if `Exec` is the best phrase here. Maybe full `ExecutionCommand`, or `Command`, or just `Execution`?

File: sdk/resourcemanager/azure-resourcemanager-containerinstance/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java [hunk 2]
```diff
@@ -922,6 +922,78 @@ WithContainerInstanceAttach<ParentT> withReadOnlyVolumeMountSetting(
...
             }
 
+            /**
+             * The stage of the container instance definition allowing to specify liveness probe.
+             * Azure Container Instances supports liveness probes so that you can configure your containers within your
+             * container group to restart if critical functionality is not working.
+             *
+             * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+             */
+            interface WithLivenessProbe<ParentT> {
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-containerinstance/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java:932]: nit, is it updatable?

File: sdk/resourcemanager/azure-resourcemanager-containerinstance/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java (hunk 1 above)
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-containerinstance/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java:940]: Should we note that only 1 liveness probe can be set (if it is the case)?

**Review by weidongxu-microsoft — APPROVED** [review]
@haolingdong-msft for review.

File: sdk/resourcemanager/azure-resourcemanager-containerinstance/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java (hunk 2 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-containerinstance/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java:932]: Tried with serviceClient, and yes it's updatable...

The one complication is that currently we don't have a way to update child container in `ContainerGroup`, i.e. no `ContainerGroup::updateContainerInstance`. Also tried some previous properties like `ports`, and it's also updatable. 

Maybe we do child container update in a separate PR?

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-containerinstance/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java:932]: sure

File: sdk/resourcemanager/azure-resourcemanager-containerinstance/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java (hunk 1 above)
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-containerinstance/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java:940]: Then probably `ExecutionCommand` or `Command`. May ref their doc https://learn.microsoft.com/en-us/azure/container-instances/container-instances-liveness-probe#liveness-command

File: sdk/resourcemanager/azure-resourcemanager-containerinstance/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java (hunk 2 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-containerinstance/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java:932]: issue logged: https://github.com/Azure/azure-sdk-for-java/issues/32624

File: sdk/resourcemanager/azure-resourcemanager-containerinstance/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java (hunk 1 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-containerinstance/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java]: Changed. Also added overload for failureThreshold.

File: sdk/resourcemanager/azure-resourcemanager-containerinstance/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java [hunk 3]
```diff
@@ -932,28 +932,64 @@ WithContainerInstanceAttach<ParentT> withReadOnlyVolumeMountSetting(
...
+                WithContainerInstanceAttach<ParentT> withLivenessProbeExecutionCommand(List<String> command, int probePeriodSeconds);
+
+                /**
+                 * Specifies the container's liveness probe to execute a given command at a given interval.
+                 * <p>After the probe command execution failure reaches the given threshold, the container will restart.</p>
+                 * <p>Only 1 liveness probe can be set for a given container.</p>
+                 *
+                 * @param command the command for the probe to execute
+                 * @param probePeriodSeconds the interval at which the command executes
+                 * @param failureThreshold the failure threshold before the container restarts
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-containerinstance/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java:953]: at least make clear in doc, what threshold? seconds? count?

Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-containerinstance/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java:953]: Makes sense. I think it's the consecutive failures in a row, paring with Kubernetes's threshold. I'll double-check. Thanks!

Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-containerinstance/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java]: Added verfication for failureThreshold. It is indeed consecutive failures in a row.

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
@haolingdong-msft Would you also take a review?

**Review by haolingdong-msft — APPROVED** [review]

**Pull request merged**